src/api/modules/sign_data_item/sign_data_item.background.ts

```diff
@@ -75,6 +81,9 @@ const background: BackgroundModuleFunction<number[]> = async (
   if (decryptedWallet.type == "local") {
     // create bundlr tx as a data entry
     const dataSigner = new ArweaveSigner(decryptedWallet.keyfile);
+    dataSigner.sign = function (message) {
```

Comment:
It probably makes no difference but I'd try to keep the code unchanged for those not using the new `signatureOptions`:

```
if (signatureOptions) {
  dataSigner.sign = (message) => getCryptoDriver().sign(this.jwk, message, signatureOptions) as any;
}
```